### ISSUE TITLE
Frozen browser?

### ISSUE THREAD

**Issue by ufb** [ufb/Textmarker#101]
A few users experienced performance issues with Textmarker and they assume that Textmarker would cause the browser to freeze. Neither me nor the users who help with testing were able to reproduce such a behavior - on the contrary Textmarker seems to be very fast and consume little resources.

That's why we need the help of friendly users who experience such bahvior and are willing to spend a few minutes for posting some details that could help with debugging. So if you think Textmarker would slow down your browsing experience significantly please comment below or contact me via email - undflybir at gmx dot de

Following infos are possibly helpful:
Detailed description of what actions lead to the slow-down/freeze
Does it only happen when Textmarker is not being paused?
Does it only happen when having opened pages with highlights on it?
Does it also happen if you have automatic script injections disabled?

Thank you!

**Comment by zsq2010** [comment]
for me，i found that：
when a tab or paged is loading，it sometimes inject the former tab。
i found this ，because i open the side bar，on top of that bar displaying the website title right?!
so i guess，fast switching，especially on heavy working of firefox browser，will cause this happen ，that when navigate to a new url，the former one，will be blocked
on my pc，it rarely happened

another situation is，sometimes i found the note is not loaded。 i will try pause ext from the browser action button menu。 after clicking the triangle button back and forth，the ext will work again。

in both situation，i guess it is associated with script load in heavy browser work

**Comment by ufb** [comment]
> so i guess，fast switching，especially on heavy working of firefox browser，will cause this happen

Yes, that's right. It was a painful experience making the sidebar tab-specific (with all the tab-specific displays and actions). The sidebar needs to update - and therefore render - each time you switch the tab, so fast switching can cause issues there.
But this is an effect that goes the other direction: "heavy working of firefox browser" causes render problems in Textmarker's sidebar. Have you experienced a negative performance impact the other way around though - that Textmarker causes the browser to slow down?

Concerning the issues you mentioned: Unfortunately I don't see a straight-forward solution for overcoming them. But maybe it would be helpful for you if you'd use the manual load option, available since 5.1.2:
In the miscellaneous settings disable automatic loading of Textmarker. Then you can use the edit icon in your URL bar to inject/load the Textmarker script into the current page. But of course, this won't prevent render-issues in the sidebar once you already loaded Textmarker into all pages and then switch between tabs rapidly. Open for suggestions..

**Comment by zsq2010** [comment]
i played well with the tool,just a little bit annoied with the flash of ui on loading
i was very thankful for this tool develpment
little problem cannot override the useful ,and great and smart effort

i am the keyboard shortcut fan,if all the contrls can be done with the shortcut,i just shortcut it
it is a reasonable to have ui icon,other user need it
and it also help me on broken keyboard shortcuts,like delete,undo,as conflicting with other extension
